### ISSUE TITLE
`eslint-plugin-unicorn(prefer-spread)` is triggering on strings

### ISSUE THREAD

**Issue by tylerlaprade** [oxc-project/oxc#5248]
```ts
const x = "foo";
x.concat(x);
```
This should not be flagged as `x` is not an array.
